cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -351,8 +351,10 @@ private CloseableIterator<StructuredRow> getScanApplicationsIterator(StructuredT
     if (latestOnly) {
       // if only return latest version of the app, whether the range uses version field or creationTime field
       // doesn't matter, since records will be sort on appName
-      return table.scan(range, Integer.MAX_VALUE,
-                        Fields.stringField(StoreDefinition.AppMetadataStore.LATEST_FIELD, "true"), sortOrder);
+      Collection<Field<?>> filterIndexes =
+        ImmutableList.of(Fields.stringField(StoreDefinition.AppMetadataStore.LATEST_FIELD, "true"),
+                         Fields.stringField(StoreDefinition.AppMetadataStore.LATEST_FIELD, null));
```

Comment:
maybe I missed something, but why are null values considered the same as latest=true? Isn't it possible to have a bunch of pre-6.8 rows where latest==null? Are we doing a data migration somewhere?

Whatever the answer is, it should probably be included in the comment above, explaining why this is correct.

Reply:
Yeah this was a discussion brought up by @sumengwang and @tivv
So for all the pre 6.8 pipelines, latest==null applies to them. We should treat them as latest versions.  It's okay to have the customized versions in the deployed list UI. After 6.8, latest="false" is the accurate non-latest versions. Will add the comment